### PR TITLE
feat: Add support for refreshing jokes

### PR DESCRIPTION
**Pull request by arnest00**
## Description
- `jokeFetcher` function accepts jokeIdx parameter
- Add tests for `getNextJokeIdx` function
- `getNextJokeIdx` function handles logic for cycling retrieving next joke and cycling back to beginning index
- Add support for refreshing jokes
  - Refresh button cycles through the first five jokes in `jokes.json`.

## Spec
See Issue: [FSA22V1-77](https://sparkbox.atlassian.net/browse/FSA22V1-77)

## Validation
- [ ] This PR has code changes, and our linters still pass.
- [ ] This PR has new code, so new tests were added or updated, and they pass.

### To Validate
1. Make sure all PR Checks have passed.
2. Pull down all related branches.
3. Install all dependencies with `npm i`.
4. Start the API server with `npm run api`.
5. Start the application with `npm run dev`.
6. Verify refresh button cycles through five jokes before starting back at the beginning.
7. Verify all tests pass.

### CODE REVIEW

File: src/js/index.js [hunk 1]
```diff
@@ -1,9 +1,33 @@
...
+  refreshButton.addEventListener('click', async (e) => {
+    e.preventDefault();
+    currentJokeState = getNextJokeIdx(currentJokeState, UPPER_BOUND);
+    addNextJoke(currentJokeState);
+  });
 };
-main();
+
+const showCurrentJoke = async () => {
+  const currentJokeIdx = 0;
```
Comment (arnest00) [inline src/js/index.js:26]: Keeing this line in so down the line, we can replace with a random initial load index.

File: src/js/getNextJokeIdx.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+export const getNextJokeIdx = (currentJokeIdx, upperBound) => {
```
Comment (changangus) [inline src/js/getNextJokeIdx.js:1]: This looks great! Another way if you're interested is using a conditional statement like this: 

`const getNextJokeIdx = (currentJokeIdx, upperBound) => (
  currentJokeIdx + 1 > upperBound ? 0 : currentJokeIdx + 1
);`

No need to change this though, just sharing!

File: src/js/getNextJokeIdx.spec.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import { getNextJokeIdx } from './getNextJokeIdx';
```
Comment (changangus) [inline src/js/getNextJokeIdx.spec.js:1]: You might want to default export the `getNextJokeIdx` function if it's the only function you plan on having in that file, then you don't have to use named imports.

File: src/js/jokeFetcher.spec.js [hunk 4]
```diff
@@ -17,13 +17,19 @@ afterEach(() => {
...
   it('fetches joke', async () => {
-    const joke = await jokeFetcher();
+    const joke = await jokeFetcher(0);
     expect(joke).toEqual({
       setup: 'Why did the chicken cross the playground?',
       punchline: 'To get to the other slide.',
     });
   });
 
+  it('makes a call to the correct API endpoint', async () => {
```
Comment (changangus) [inline src/js/jokeFetcher.spec.js:27]: Your tests look solid!

**Review by changangus — APPROVED** [review]

**Pull request merged**